### PR TITLE
Parishes Endpoint

### LINKED ISSUE #12: Parishes Endpoint
**Description**
Create all the classes and functions needed to access the `GET` endpoint `{{base_url}}/parishes`

**File Location**
```
- data
|__ remote
- domain
|_ mamagers
```

**Requirements**
- A `endpoints` class where the *endpoint* is declared must be created
- A `services` class where the mapping from `Response` to `Parish` must be created

*Example Response*

```
{
    "links": {
        "first": "https://api.vost.pt/v1/parishes?page=1",
        "last": "https://api.vost.pt/v1/parishes?page=3092",
        "next": "https://api.vost.pt/v1/parishes?page=2"
    },
    "data": [
        {
            "type": "parishes",
            "id": "3086",
            "attributes": {
                "code": "480106",
                "name": "LOMBA",
                "created_at": "2019-07-02 13:31:18",
                "updated_at": "2019-07-02 13:31:18"
            },
            "links": {
                "self": "https://api.vost.pt/v1/parishes/3086"
            }
        }
    ],
    "meta": {
        "items": 1,
        "total": 3092
    }
}
```

### PR DESCRIPTION
**Pull request by Vanethos**
Add Parish Endpoint
Add Parish Models
Add Parish Mappers
Add Parish fetch from server to home, displaying all data in a list

closes #12

### CODE REVIEW

File: lib/data/remote/endpoints/parish_endpoints.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import 'package:dio/dio.dart';
+
+import '../../../constants.dart';
+
+const todoEndpoint = "todos/1";
```
Comment (OldMetalmind) [inline lib/data/remote/endpoints/parish_endpoints.dart]: For what are you using this?

File: lib/data/remote/models/response/attribute_response.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+import 'package:vost/data/remote/models/_base/parser.dart';
+
+part 'attribute_response.g.dart';
+
+abstract class AttributeResponse implements Built<AttributeResponse, AttributeResponseBuilder>, SerializedModel<AttributeResponse> {
```
Comment (OldMetalmind) [inline lib/data/remote/models/response/attribute_response.dart]: My suggestion is to follow the same structured used in the VOST API and have the Attribute object to include the following fields:
```
    attributes:
      type: object
      properties:
        locality:
          type: string
        latitude:
          type: number
          format: double
        longitude:
          type: number
          format: double
        started_at:
          type: string
          format: date-time
        ended_at:
          type: string
          format: date-time
        created_at:
          type: string
          format: date-time
        updated_at:
          type: string
          format: date-time
```

Given most are optional, nullable.

File: lib/domain/models/attribute_model.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import 'package:built_value/built_value.dart';
+
+part 'attribute_model.g.dart';
+
+abstract class AttributeModel implements Built<AttributeModel, AttributeModelBuilder> {
```
Comment (OldMetalmind) [inline lib/domain/models/attribute_model.dart:5]: Please check my comment for `AttributeResponse`

**Review by OldMetalmind — COMMENTED** [review]
I've let some comments that I would like to be discussed, minor questions.

File: lib/data/remote/models/response/attribute_response.dart (hunk 2 above)
Comment (Vanethos) [reply lib/data/remote/models/response/attribute_response.dart]: What about the `code` and the `name`? Both were in the endpoint response but aren't specified here

Comment (OldMetalmind) [reply lib/data/remote/models/response/attribute_response.dart]: That was my bad, those should be included, I was trying to find a `set` of attributes that include all fields used everywhere.

**Review by OldMetalmind — APPROVED** [review]

**Pull request merged**